src/docker/compose.local.yaml

```diff
@@ -1,51 +1,63 @@
 # yamllint disable rule:empty-lines rule:key-ordering rule:brackets
+# cspell:word localnet
 ---
 services:
   aptos-node:
     network_mode: 'host'
-    build:
-      context: '../../'
-      dockerfile: 'src/docker/aptos-node/Dockerfile'
+    image: 'econialabs/aptos-cli:4.1.0'
+    profiles: ["test-harness"]
     ports:
     - '50051:50051'
     - '8070:8070'
     - '8090:8090'
     - '8080:8080'
     - '8081:8081'
     healthcheck:
-      test: ['CMD', 'curl', '-f', 'http://localhost:8070/']
-      interval: '30s'
-      timeout: '10s'
-      retries: 5
-      start_period: '30s'
-    command: ['bash', '/app/run-fullnode.sh']
+      test: ['CMD', 'curl', 'http://localhost:8070/']
+      interval: '5s'
+      timeout: '5s'
+      retries: 10
+      start_period: '60s'
     # In order to start the node with the indexer API; i.e., run
-    # `aptos node run-local-testnet --with-indexer-api`, the Docker socket must
+    # `aptos node run-localnet --with-indexer-api`, the Docker socket must
     # be mounted to the container. This mount enables the container to start a
-    # new container. Note that this is *not* "Docker-in-Docker", but rather
-    # "Docker-outside-Docker". That is, the container can act as the host in
+    # new container. Note that this is *not* 'Docker-in-Docker', but rather
+    # 'Docker-outside-Docker'. That is, the container can act as the host in
     # terms of creating and orchestrating other containers but still appears as
     # a sibling to the existing containers and any new containers it creates.
     volumes:
     - '/var/run/docker.sock:/var/run/docker.sock'
-
-  move:
+    entrypoint: [
+      'aptos',
+      'node',
+      'run-localnet',
+      '--with-indexer-api',
+      '--force-restart',
+      '--bind-to',
+      '0.0.0.0'
+    ]
+  local-deployer:
     extra_hosts:
     - 'host.docker.internal:host-gateway'
     build:
-      context: '../../'
-      dockerfile: 'src/docker/aptos-node/Dockerfile'
-    environment:
-      PUBLISHER_PK: '${PUBLISHER_PK}'
-      BIG_MONEY_GUY_PK: '${BIG_MONEY_GUY_PK}'
-      EMOJICOIN_MODULE_ADDRESS: '${EMOJICOIN_MODULE_ADDRESS}'
+      context: ../../
+      dockerfile: 'src/docker/local-deployer/Dockerfile'
+      args:
+        PUBLISHER_PRIVATE_KEY: '${PUBLISHER_PRIVATE_KEY}'
+        EMOJICOIN_MODULE_ADDRESS: '${EMOJICOIN_MODULE_ADDRESS}'
+    env_file:
+    - path: './example.local.env'
+      required: true
+    - path: './.env'
+      required: false
+    image: 'econialabs/emojicoin-dot-fun-local-deployer:latest'
+    profiles: ["test-harness"]
```

Comment:
This is a very interesting setting that I didn't know about. How about merging the two compose files and using profiles instead ? Deploying the indexing stack and pointing it to testnet would look something like `docker compose --profile testnet up -d` and running locally would look like `docker compose --profile local up -d`.

Reply:
+1 ^

Reply:
They're nice to use in terms of excluding/including specific services, but actually orchestrating the order of services is a bit of a pain...I used a `frontend` profile because it's really the easiest to manage, but using more than that didn't really make sense.

It's all in the `README.md` now, too!